overlord/state/task.go

```diff
@@ -194,20 +198,16 @@ func (t *Task) Log() []string {
 	return t.log
 }
 
-// Logf logs textual information about the progress of the task.
-// Only the most recent entries logged are held in memory, potentially
-// with different behavior for different task statuses. How many entries
-// are held is an implementation detail and may change over time.
+// Logf logs information about the progress of the task.
 func (t *Task) Logf(format string, args ...interface{}) {
 	t.state.ensureLocked()
 	t.addLog(LogInfo, format, args)
 }
 
-// Errorf sets the task to ErrorStatus and logs the message as an error.
+// Errorf logs error information about the progress of the task.
```

Comment:
Agreed.. I also don't want to rule that out just yet. It may turn out to be useful to tag non-final errors message as such (IOW, call Errorf while Task.Status() != ErrorStatus).